packages/flutter/lib/src/widgets/editable.dart

```diff
@@ -430,3 +463,153 @@ class _EditableLineWidget extends LeafRenderObjectWidget {
     return new TextSpan(style: style, text: text);
   }
 }
+
+enum _TextSelectionHandleSide { left, right }
+
+class _TextSelectionHandleOverlay extends StatefulWidget {
+  _TextSelectionHandleOverlay({
+    Key key,
+    this.onSelectionHandleChanged,
+    this.selection,
+    this.renderObject
+  }) : super(key: key);
+
+  final TextSelection selection;
+  final ValueChanged<TextSelection> onSelectionHandleChanged;
+  // TODO(mpcomplete): what if the renderObject is removed or replaced, or
+  // moves? Not sure what cases I need to handle, or how to handle them.
+  final RenderEditableLine renderObject;
+
+  @override
+  _TextSelectionHandleOverlayState createState() => new _TextSelectionHandleOverlayState();
+}
+
+/// The state for an overlay that displays 2 selection handles. This is intended
+/// to be used only when there is text selected (e.g. the TextSelection is not
+/// collapsed).
+class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay> {
+  TextSelection selection;
+  List<Point> endpoints;
+
+  @override
+  void initState() {
+    super.initState();
+    selection = config.selection;
+    endpoints = config.renderObject.getEndpointsForSelection(selection);
+  }
+
+  @override
+  void didUpdateConfig(_TextSelectionHandleOverlay oldConfig) {
+    setState(() {
+      selection = config.selection;
+      endpoints = config.renderObject.getEndpointsForSelection(selection);
```

Comment:
Why not call this during build?  There doesn't seem to be any advantage in caching the information since you read it both in `initState` and in `didUpdateConfig`.